miden-lib/src/tests/test_account.rs

```diff
@@ -326,13 +326,100 @@ fn test_set_item() {
     let _process = run_tx(&transaction).unwrap();
 }
 
-// TODO: reenable once storage map support is implemented
-#[ignore]
+// Test different account storage types
+#[test]
+fn test_get_storage_data_type() {
+    for storage_item in [storage_item_0(), storage_item_1(), storage_item_2()] {
+        let (tx_inputs, tx_args) =
+            mock_inputs(MockAccountType::StandardExisting, AssetPreservationStatus::Preserved);
+
+        let code = format!(
+            "
+        use.miden::kernels::tx::account
+        use.miden::kernels::tx::prologue
+
+        begin
+            # prepare the transaction
+            exec.prologue::prepare_transaction
+
+            # push the account storage item index
+            push.{item_index}
+
+            # get the data type of the respective storage slot
+            exec.account::get_storage_slot_type_info
+
+        end
+        ",
+            item_index = storage_item.index,
+        );
+
+        let transaction =
+            prepare_transaction(tx_inputs.clone(), tx_args.clone(), code.as_str(), None);
+        let _process = run_tx(&transaction).unwrap();
+        let transaction = prepare_transaction(tx_inputs, tx_args, &code, None);
+        let process = run_tx(&transaction).unwrap();
+
+        let storage_slot_data_type = match storage_item.slot.slot_type {
+            StorageSlotType::Value { value_arity } => (value_arity, 0),
+            StorageSlotType::Map { value_arity } => (value_arity, 1),
+            StorageSlotType::Array { value_arity, depth } => (value_arity, depth),
+        };
+
+        assert_eq!(process.get_stack_item(0), Felt::from(storage_slot_data_type.0));
+        assert_eq!(process.get_stack_item(1), Felt::from(storage_slot_data_type.1));
```

Comment:
IMO we should compare the process complete stack against the expected values. This is because:

- I have found code that doesn't properly clean the stack, and the tests don't catch it because it only asserts on the expected output, but not that the rest of the stack is clean (by clean, I mean the stack should be padded to 16 elements and the elements on the top of the stack should be zero)
- I have found many tests hard to understand, specially when it is something I have not written and starts to fail because a change I made broke a test assumption. When this happens I want to see the complete stack output, so that I can see if words are swapped, or if there is additional data I was not expecting.

Given the above, I think this should instead:

- construct an array of felts, representing the top 16 elements, and check the process return stack matches that
- check the process stack is exactly 16 elements, and there are not more elements after it (i.e. no overflow)

Reply:
ok, that makes sense

Reply:
OK, in this test this doesn't make sense. 

Here there is no padding. We call the account kernel procedure directly. There is no `call` or `syscall` involved. The `kernel::tx::account.masm` procedures don't prepare the stack for return, because they are called directly in the kernel context. 

I will add this to the next test though, where we call procedures from the account context.